### PR TITLE
re-enable codecov

### PR DESCRIPTION
**Pull request by rochdev**
### What does this PR do?

Re-enable codecov.

### Motivation

It was disabled a while ago because reports were timing out and it was unclear whether CircleCI or Codecov was at fault. We need code coverage reports so it should be re-enabled. If we are still having the issue, we should work with them to figure out why.

### CODE REVIEW

**Review by bengl — APPROVED** [review]

**Comment by rochdev** [comment]
Let's not merge this for now since it seems that the codecov reporter is flaky. I have opened a support ticket with codecov to sort this out.

**Comment by rochdev** [comment]
The error was actually the API rate limit being hit. This was because we were sending reports from all jobs individually so that they are merged in the codecov UI. I updated the build to merge before sending everything as a single report to avoid hitting the rate limit.

@bengl This is now ready for review.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1111,6 +1141,63 @@ workflows:
       - node-tedious
       - node-when
       - node-winston
+      - codecov:
+          requires:
+            - node-core-8
```
Comment (bengl) [inline .circleci/config.yml:1146]: This whole list is repeated 3 or 4 times in this file. Can DRY it up?

**Review by bengl — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (rochdev) [reply .circleci/config.yml:1146]: Doesn't look like it unfortunately as the idea was [rejected](https://github.com/yaml/yaml/issues/35).

**Review by bengl — APPROVED** [review]

**Pull request merged**